server/auth.go

```diff
@@ -132,6 +135,26 @@ func (s *Server) isRouterAuthorized(c *client) bool {
 	return comparePasswords(opts.Cluster.Password, c.opts.Password)
 }
 
+// removeUnauthorizedSubs removes any subscriptions the client has that are no
+// longer authorized, e.g. due to a config reload.
+func (s *Server) removeUnauthorizedSubs(c *client) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if c.perms == nil {
+		return
+	}
+
+	for sid, sub := range c.subs {
+		if !c.canSubscribe(sub.subject) {
+			delete(c.subs, sid)
```

Comment:
Yeah I think so. The protocol documentation needs to be updated though:
```
-ERR	Server	Indicates a protocol error. Will cause client disconnect.
```

https://nats.io/documentation/internals/nats-protocol/